test/Broadway/EventStore/Management/EventStoreManagementTest.php

```diff
@@ -0,0 +1,348 @@
+
+abstract class EventStoreManagementTest extends TestCase
+{
+    /**
+     * @var EventStoreInterface|EventStoreManagementInterface
+     */
+    protected $eventStore;
+
+    /**
+     * @var DateTime
+     */
+    protected $now;
+
+    public function setUp()
+    {
+        $this->now = DateTime::now();
+        $this->eventStore = $this->createEventStore();
+        $this->createAndInsertEventFixtures();
+        $this->eventVisitor = new RecordingEventVisitor();
+    }
+
+    protected function visitEvents(Criteria $criteria = null)
+    {
+        $eventVisitor = new RecordingEventVisitor();
+
+        $this->eventStore->visitEvents($eventVisitor, $criteria);
+
+        return $eventVisitor->getVisitedEvents();
+    }
+
+    abstract protected function createEventStore();
+
+    /** @test */
+    public function it_visits_all_events()
+    {
+        $visitedEvents = $this->visitEvents();
+
+        $this->assertEquals($this->getEventFixtures(), $visitedEvents);
+    }
+
+    /** @test */
+    public function it_visits_aggregate_root_id()
+    {
+        $visitedEvents = $this->visitEvents(Criteria::create()->withAggregateRootId(
+            $this->getId(1)
+        ));
+
+        $this->assertEquals(array(
+            $this->createDomainMessage(1, 0, new Start()),
+            $this->createDomainMessage(1, 1, new Middle('a')),
+            $this->createDomainMessage(1, 2, new Middle('b')),
+            $this->createDomainMessage(1, 3, new Middle('c')),
+            $this->createDomainMessage(1, 4, new Middle('d')),
+            $this->createDomainMessage(1, 5, new End()),
+        ), $visitedEvents);
+    }
+
+    /** @test */
+    public function it_visits_aggregate_root_ids()
+    {
+        $visitedEvents = $this->visitEvents(Criteria::create()->withAggregateRootIds(array(
+            $this->getId(1),
+            $this->getId(3),
+        )));
+
+        $this->assertEquals(array(
+            $this->createDomainMessage(1, 0, new Start()),
+            $this->createDomainMessage(1, 1, new Middle('a')),
+            $this->createDomainMessage(1, 2, new Middle('b')),
+            $this->createDomainMessage(1, 3, new Middle('c')),
+            $this->createDomainMessage(3, 0, new Start()),
+            $this->createDomainMessage(3, 1, new Middle('a')),
+            $this->createDomainMessage(3, 2, new Middle('b')),
+            $this->createDomainMessage(3, 3, new Middle('c')),
+            $this->createDomainMessage(1, 4, new Middle('d')),
+            $this->createDomainMessage(3, 4, new Middle('d')),
+            $this->createDomainMessage(1, 5, new End()),
+            $this->createDomainMessage(3, 5, new End()),
+        ), $visitedEvents);
+    }
+
+    /** @test */
+    public function it_visits_additional_aggregate_root_ids()
+    {
+        $visitedEvents = $this->visitEvents(Criteria::create()
+            ->withAggregateRootId($this->getId(1))
+            ->withAdditionalAggregateRootId($this->getId(3))
```

Comment:
this is mostly testing the `Criteria` DSL. I'd say a test for 1 and multiple aggregate root ids is fine.